src/test/java/org/elasticsearch/test/ElasticsearchTestCase.java

```diff
@@ -168,7 +168,20 @@ public static boolean awaitBusy(Predicate<?> breakPredicate) throws InterruptedE
     }
 
     public static boolean awaitBusy(Predicate<?> breakPredicate, long maxWaitTime, TimeUnit unit) throws InterruptedException {
+        return awaitBusy(breakPredicate, maxWaitTime, 0, unit);
+    }
+
+    /**
+     * Repeat a task until it returns true or after a given wait time.
+     * By default, it will increase by 2 the sleep time between two runs
+     * @param breakPredicate test you want to run
+     * @param maxWaitTime maximum time you want to wait
+     * @param maxSleepTime if >0 maximum sleep time before two runs
+     * @param unit time unit used for maxWaitTime and maxSleepTime
+     */
+    public static boolean awaitBusy(Predicate<?> breakPredicate, long maxWaitTime, long maxSleepTime, TimeUnit unit) throws InterruptedException {
         long maxTimeInMillis = TimeUnit.MILLISECONDS.convert(maxWaitTime, unit);
+        long maxSleepTimeInMillis = TimeUnit.MILLISECONDS.convert(maxSleepTime, unit);
         long iterations = Math.max(Math.round(Math.log10(maxTimeInMillis) / Math.log10(2)), 1);
```

Comment:
@jpountz Actually, my PR might be wrong here as I did not change this `iterations` computation. Any idea what the value should be?